### DISCUSSION TITLE
replacing-ajax-calls-in-razor-pages-using-razor-components-and-blazor/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#109] [Announcements]
# replacing-ajax-calls-in-razor-pages-using-razor-components-and-blazor/

In this post I show how you can get AJAX-like behaviour in your Razor Pages by using Razor Components and Blazor

https://andrewlock.net/replacing-ajax-calls-in-razor-pages-using-razor-components-and-blazor/

**Comment by Majestic-Mustang** [comment]
Great blog. Thank you!

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 16 2020, 12:12, in reply to [Göran Halvarsson](https://disqus.com/by/granhalvarsson/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/109#discussioncomment-6058446)</em>

---

<p>Good question :) The advantage would be that the rest of the form works in a disconnected way, so if network drops, the page won't freeze. But I agree, that's no much of an argument!</p>

**Comment by lradunovic** [comment]
This example does not work. CountrySelectionChanged never gets triggered.
